Etabs_Adapter/CRUD/Read/SectionProperty.cs

```diff
@@ -0,0 +1,292 @@
+                    case eFramePropType.CoreBrace_BRB:
+                        break;
+                    case eFramePropType.ConcreteTee:
+                        m_model.PropFrame.GetConcreteTee(id, ref fileName, ref materialName, ref t3, ref t2, ref tf, ref tw, ref twt, ref verticalFlip, ref colour, ref notes, ref guid);
+                        dimensions = Engine.Geometry.Create.TSectionProfile(t2, t2, tw, tf, 0, 0, verticalFlip);
+                        break;
+                    case eFramePropType.ConcreteBox:
+                        m_model.PropFrame.GetTube(id, ref fileName, ref materialName, ref t3, ref t2, ref tf, ref tw, ref colour, ref notes, ref guid);
+                        if (tf == tw)
+                            dimensions = Engine.Geometry.Create.BoxProfile(t3, t2, tf, 0, 0);
+                        else
+                            dimensions = Engine.Geometry.Create.FabricatedBoxProfile(t3, t2, tw, tf, tf, 0);
+                        break;
+                    case eFramePropType.ConcretePipe:
+                        m_model.PropFrame.GetPipe(id, ref fileName, ref materialName, ref t3, ref tw, ref colour, ref notes, ref guid);
+                        dimensions = Engine.Geometry.Create.TubeProfile(t3, tw);
+                        break;
+                    case eFramePropType.ConcreteCross:
+                        break;
+                    case eFramePropType.SteelPlate:
+                        m_model.PropFrame.GetPlate(id, ref fileName, ref materialName, ref t3, ref t2, ref colour, ref notes, ref guid);
+                        dimensions = Engine.Geometry.Create.RectangleProfile(t3, t2, 0);
+                        break;
+                    case eFramePropType.SteelRod:
+                        m_model.PropFrame.GetRod(id, ref fileName, ref materialName, ref t3, ref colour, ref notes, ref guid);
+                        dimensions = Engine.Geometry.Create.CircleProfile(t3);
+                        break;
+                    default:
+                        break;
+                }
+                if (dimensions == null)
+                {
+                    Engine.Reflection.Compute.RecordNote(propertyType.ToString() + " properties are not implemented in ETABS adapter. An empty section has been returned.");
+                    constructSelector = "explicit";
+                }
+                #endregion
+
+
+                IMaterialFragment material;
+                try
+                {
+                    material = bhomMaterials[materialName];
+                }
+                catch (Exception)
+                {
+                    material = bhomMaterials.FirstOrDefault().Value;
+                    Engine.Reflection.Compute.RecordNote("Could not get material from ETABS. Using a default material");
+                }
+
+
+                switch (constructSelector)
+                {
+                    case "fromDimensions":
+                        if (material is Steel)
```

Comment:
Here you could just use the `SectionProeprtyFromProfile` method instead and skip the whole switch statement (that method does that for you)